File: Lab9/Shoes.py
class Shoes:
    def __init__(self, name, size, material):
        self.is_put_on = False
        self.size = size
        self.material = material
        self.lacing_types = ["Hash", "Twistie", "Riding bow", "Lattice", "Zipper"]
        self.lacing = "Hash"
        self.washed = Wash()
        self.repaired = Repair()
        self.name = name
        self.count = 0

        if size > 37:
            self.count += 1

    def start(self):
        if not self.is_put_on:
            self.is_put_on = True
            self.washed.weared()
            self.repaired.weared()
            print("Put on")
        else:
            print("You already put on these shoes")

    def end(self):
        if self.is_put_on:
            self.is_put_on = False
            print("Take off")
        else:
            print("You took off those shoes")

    def show_lacing(self):
        print(f"{self.lacing} lacing")

    def show_material(self):
        print(f"Your shoes are made of {self.material}")

    def change_lacing(self, lacing_type):
        self.lacing = self.lacing_types[lacing_type - 1]
        print(f"The lacing type changed to: {self.lacing}")

    def show_size(self):
        print(f"Shoe size is: {self.size}")

    def show_info(self):
        print(f"Name: {self.name} ; Size: {self.size} ; Material: {self.material}")

    def show_clean(self):
        print(self.washed.is_washed())

    def clean(self):
        return self.washed.clean()

    def show_repair(self):
        print(self.repaired.is_repaired())

    def repair(self):
        return self.repaired.repair1()

    def dispose(self):
        pass  # Release any resources

class Wash:
    def __init__(self):
        self.washed = 100

    def is_washed(self):
        return f"Your shoes are {self.washed}% clean"

    def clean(self):
        self.washed = 100
        return "Your shoes are now clean"

    def weared(self):
        self.washed -= 2
        if self.washed <= 0:
            self.washed = 0

class Repair:
    def __init__(self):
        self.repaired = 100

    def is_repaired(self):
        return f"Your shoes are {self.repaired}% unharmed"

    def repair1(self):
        self.repaired = 100
        return "Your shoes have been repaired"

    def weared(self):
        self.repaired -= 1
        if self.repaired <= 0:
            self.repaired = 0
            print("Your shoes are beyond repair :(")

File: Lab2/src/KI305/Nikolenko/Lab2/Shoes.java
package KI305.Nikolenko.Lab2;
import java.io.*;
import java.util.List;
import java.util.ArrayList;
/**
 * Class <code>Shoes</code>
 * @author Oleksandr Nikolenko
 * @version 1.0
 */

public class Shoes {
    boolean isPutOn = false;
    int size = 0;
    String material;
    String [] lacingTypes = { "Hash", "Twistie", "Riding bow", "Lattice","Zipper" };
    String lacing = "";
    Wash washed = new Wash();
    Repair repaired = new Repair();
    private PrintWriter fout;
    /**
     * Constructor
     * Creates shoes pair
     */

    public static List<String> getMaterialsList() {
        return materialsList;
    }

    private static List<String> materialsList = new ArrayList<>();

    /**
     * Method for put on shoes
     */
    public void Start(){
        if (isPutOn != true){
            isPutOn = true;
            washed.weared();
            repaired.weared();
            System.out.println("Put on");
            fout.println("Put on");
        }
        else{
            System.out.println("You already put on these shoes");
            fout.println("You already put on these shoes");

        }
    }

    /**
     * Method for taking off the shoes
     */
    public void End(){
        if (isPutOn != false){
            isPutOn = false;
            System.out.println("Take off");
            fout.println("Take off");
        }
        else{
            System.out.println("You took off those shoes");
            fout.println("You took off those shoes");
        }
    }

    /**
     * Method shows lacing type
     */
    public void showLacing(){
        System.out.println(lacing+" lacing");
        fout.println(lacing+" lacing");
    }
    /**
     * Method shows material type
     */
    public void showMaterial(){
        System.out.println("Youre shoes are made of-"+material);
        fout.println("Youre shoes are made of-"+material);
    }
    /**
     * Method changes lacing type
     */
    public void changeLacing(int type){
        lacing = lacingTypes[type-1];
        System.out.println("The lacing type changed to: "+lacing);
        fout.println("The lacing type changed to: "+lacing);
    }

    /**
     * Method shows size
     */
    public void showSize(){
        System.out.println("Shoe size is:"+size);
        fout.println("Shoe size is:"+size);
    }

    /**
     * Method shows if cleaning needed
     */
    public void showClean(){
        System.out.println(washed.isWashed());
        fout.println(washed.isWashed());
    }
    /**
     * Method cleans the shoes
     */
    public void clean(){
        System.out.println(washed.clean());
        fout.println(washed.clean());
    }
    /**
     * Method shows if repair is needed
     */
    public void showRepair(){
        System.out.println(repaired.isRepaired());
        fout.println(repaired.isRepaired());
    }
    /**
     * Method repairs the shoes
     */
    public void repair(){
        System.out.println(repaired.Repair1());
        fout.println(repaired.Repair1());
    }
    /**
     * Method releases used recourses
     */
    public void dispose()
    {
        fout.flush();
        fout.close();
    }
    public Shoes(int s, String m) throws FileNotFoundException {
        lacing = "Hash";
        size = s;
        material = m;
        fout = new PrintWriter(new File("KI305.Nikolenko.Lab2.txt"));

        // Додайте матеріал до списку, якщо розмір більше або рівний 40
        if (size >= 40) {
            materialsList.add(material);
        }
    }

}

